integration/spark2/src/main/java/org/apache/carbondata/datamap/PreAggregateDataMapProvider.java

```diff
@@ -77,13 +79,15 @@ private void validateDmProperty(DataMapSchema dataMapSchema)
   }
 
   @Override
-  public void cleanMeta() {
+  public void cleanMeta() throws IOException {
+    DataMapSchema dataMapSchema = getDataMapSchema();
     dropTableCommand = new CarbonDropTableCommand(
         true,
         new Some<>(dbName),
         tableName,
         true);
     dropTableCommand.processMetadata(sparkSession);
+    DataMapStoreManager.getInstance().dropDataMapSchema(dataMapSchema.getDataMapName());
```

Comment:
Can you check how MV deal with this? In another word, what's the logic of dropping a MV?

Reply:
when create a mv datamap, will create a datamap scheme file and register a datamap catalog; then when drop the mv datamap,will delete the datamap schema file and unregister the datamap catalog.